src/asynciterable/operators/union.ts

```diff
@@ -54,7 +56,7 @@ export function union<TSource>(
   right: AsyncIterable<TSource>,
   comparer: (x: TSource, y: TSource) => boolean | Promise<boolean> = comparerAsync
 ): MonoTypeOperatorAsyncFunction<TSource> {
-  return function unionOperatorFunction(left: AsyncIterable<TSource>): AsyncIterableX<TSource> {
-    return new UnionAsyncIterable<TSource>(left, right, comparer);
+  return function unionOperatorFunction(left) {
```

Comment:
Doesn't using type inference here change the return signature from `AsyncIterableX<TSource>` to `UnionAsyncIterable`?

Reply:
It 	does, but only for `unionOperatorFunction`. The outer `union` (the actual API) still correctly returns `MonoTypeOperatorAsyncFunction` without leaking 🙂